### ISSUE TITLE
Date filter does not work if a date format is defined at the column

### ISSUE THREAD

**Issue by fs86** [inovua/reactdatagrid#70]
Tested with the first example at https://reactdatagrid.io/docs/filtering. Open the source code and add a date format for the column "birthDate":

```js
 {
    name: 'birthDate',
    header: 'Bith date',
    defualtFlex: 1,
    minWidth: 200,
    dateFormat: 'MM-DD-YYYY', // This like breaks the date filter logic.
    filterEditor: DateFilter,
    filterEditorProps: (props, { index }) => {
      return {
        dateFormat: 'MM-DD-YYYY',
        cancelButton: false,
        highlightWeekends: false,
        placeholder: index == 1 ? 'Created date is before...': 'Created date is after...'
      }
    },
    render: ({ value, cellProps }) => {
      return moment(value).format('MM-DD-YYYY')
    }
  },
```
Then select today's date and choose "before".

Expected behavior: Filter returns all records.
Actual behavior: Filter returns no recods.

Aside from the bug I would like to understand why I need to define a date format at column level when it is already defined via dateEditorProps.

It would also be nice if you can add the column level dateFormat property to the type definitions.

**Comment by KatyMan** [comment]
I am having the same issue when trying to filter a date column.  My date value is defined as "2021-03-25T14:05:43.751-05:00".  My dateFormat is set to "MM-DD-YY".  My renderer is formatting the date as "MM-DD-YY HH:MM A".  I thought that might be causing the issue, so I change my renderer to output moment(value).format(dateFormat), but it still behaves in the same manner as **fs86** has reported (i.e. I select a valid "after" filter date which should have selected a few records, but none are selected.  Same goes for entering a valid "before" date filter date. No rows are displayed

I am using 4.1.2 of the community version.

**Comment by radamesh** [comment]
I know the question has nothing to do with it but ...
how do I put the calendar to the left side? when i click to editable situation and open de calendar.

https://stackoverflow.com/questions/68026857/react-datagrid-datefilter-how-do-i-open-calendar-to-the-left-side
tks

**Comment by inovua-admin** [comment]
Fixed in `v4.1.24`
